### PR TITLE
fix: territories metadata type handling

### PR DESCRIPTION
**Pull request by zokito**
## What does this pull request contains

---

- [ ] Added for new features.
- [x] Changed for changes in existing functionality.
- [ ] Deprecated for soon-to-be removed features.
- [ ] Removed for now removed features.
- [x] Fixed for any bug fixes.
- [ ] Security in case of vulnerabilities.

## Explain your changes

---
Added fixes which enable the proper generation of delta package.xml files for Territory2 related metadata - nested under territory2Model folders, similar as for object folders. Next to changing the code to support it metadata json files were also changed to match the structure as demonstrated in the object definition. I've also added a functional test to verify the change made.

## Does this close any currently open issues?

---

closes https://github.com/scolladon/sfdx-git-delta/issues/206

- [x] Jest test to check the fix is applied are added.

## Any particular element to being able to test locally

---

## Any other comments?

---

Pretty cool plug-in

## Where has this been tested?

---

**Operating System:** Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64

**yarn version:** 1.22.18
**node version:** v17.4.0

**git version:** 2.23.0

**sfdx version:** sfdx-cli/7.142.1 darwin-x64 node-v16.14.0

**sgd plugin version:** sfdx-git-delta 5.0.6 (link) .../sgd/sfdx-git-delta

### CODE REVIEW

**Comment by zokito** [comment]
Hi @scolladon can you please help with the pull request lint failure?

**Comment by scolladon** [comment]
Sure , this linting issue is expected, it is because the pull request comes from a fork and does not have the rights to execute the underlying GitHub action we are using (for security reasons)

I'll make sure the title is parsable by our releaser when I'll do the full review of the PR (probably tonight)

File: __tests__/functional.test.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ const lines = [
   'A      force-app/main/default/documents/Added/doc.document',
   'A      force-app/main/default/lwc/Added/component.js`',
   'D      force-app/main/sample/objects/Account/fields/changed.field-meta.xml',
+  'A      force-app/main/default/territory2Models/EU/territories/France.territory2-meta.xml',
+  'A      force-app/main/default/territory2Models/EU/rules/France.territory2Rule-meta.xml',
```
Comment (scolladon) [inline __tests__/functional.test.js:30]: **praises:** nice way to have a red test during fixing

**Review by scolladon — CHANGES_REQUESTED** [review]
Nice PR and great job so far! Thanks a lot

Few **remarks** that should be addressed
One **suggestion** you could treat if you'd like

I think here the application also requires a Unit test for the `typeUtils.js getType()` method and I think it is the right time to add it (it should have been done long ago), could you implement it ? I think testing both having sub types and not having sub types can be enough to detect regression so far.
You can look at other test to find a way to get a metadata file in tests

I'm open to discussion, that's what code reviews are for !

**Comment by scolladon** [comment]
Hi @zokito !

Have you been able to consider my code review ?

If you do not have time, I can contribute in this PR, just give me access to the fork!

**Review by scolladon — APPROVED** [review]
LGTM

**Review by mehdisfdc — APPROVED** [review]
LGTM

**Pull request merged**